### PR TITLE
Flux oci support 7: unit tests for OCI support in flux plugin

### PR DESCRIPTION
**Pull request by gfichtenholt**
This PR is mostly about adding unit tests for OCI support in flux plugin. The unit tests allow me to test the repo and chart redis cache are working as expected.
Other things in this PR:
  - added stub for ShutDownWithDrain to rate_limiting_queue per https://github.com/vmware-tanzu/kubeapps/pull/5123#issuecomment-1194756900. I will implement once we upgrade k8s.io/client-go as stated and I can test the scenario
  - cleaned up automation login in testdata/kind-cluster-setup.sh used to set up some integration scenarios
  - bumped fluxv2 version to latest 
  - added OCI_TERMINOLOGY.md to try and explain different terminology used by upstream
  - renamed a few types, like OCIRegistry -> OCIChartRepository to be a little more consistent with existing terminology

All tests (integration+unit) passing.

### CODE REVIEW

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/OCI_TERMINOLOGY.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# OCI TERMINOLOGY
```
Comment (antgamdia) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/OCI_TERMINOLOGY.md:1]: This is great, and I'm really fine with this file being at this place for now, however, for leveraging the website docs rendering, we should move it eventually to `site/content/docs/latest/reference/developer`.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/rate_limiting_queue.go [hunk 2]
```diff
@@ -273,6 +273,26 @@ func (q *Type) ShuttingDown() bool {
...
+// the shut down has been initiated, if that is not the case: this will block
+// indefinitely. It is, however, safe to call ShutDown after having called
+// ShutDownWithDrain, as to force the queue shut down to terminate immediately
+// without waiting for the drainage.
+//
+// TODO (gfichtenholt) put a stub in here just so kubeapps can upgrade to
+// k8s.io/client-go >= 0.23.X
+// ref https://github.com/vmware-tanzu/kubeapps/pull/5123#issuecomment-1194756900
+// I will implement once we upgrade k8s.io/client-go as stated and I can test the scenario
+func (q *Type) ShutDownWithDrain() {
```
Comment (antgamdia) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/rate_limiting_queue.go:290]: Thanks! It's great as the PR we opened in Helm has been accepted, so it is a matter of time before we can finally bump up the version.

**Review by antgamdia — APPROVED** [review]
Awesome! Thoroughly testing the OCI support, that's great. I wish we had these test suites periodically executed as part of our CI as well.

**Comment by gfichtenholt** [comment]
> Awesome! Thoroughly testing the OCI support, that's great. I wish we had these test suites periodically executed as part of our CI as well.

Sure! This can happen with a little bit of work. Just a matter of prioritizing it over other things

**Pull request merged**